libraries/Microsoft.Bot.Connector/Authentication/JwtTokenValidation.cs

```diff
@@ -62,22 +62,28 @@ public static async Task<ClaimsIdentity> AuthenticateRequest(IActivity activity,
             if (string.IsNullOrWhiteSpace(authHeader))
             {
                 var isAuthDisabled = await credentials.IsAuthenticationDisabledAsync().ConfigureAwait(false);
-                if (isAuthDisabled)
+                if (!isAuthDisabled)
                 {
-                    // In the scenario where Auth is disabled, we still want to have the
-                    // IsAuthenticated flag set in the ClaimsIdentity. To do this requires
-                    // adding in an empty claim.
-                    return new ClaimsIdentity(new List<Claim>(), "anonymous");
+                    // No Auth Header and Auth is required. Request is not authorized.
+                    throw new UnauthorizedAccessException();
                 }
 
-                // No Auth Header. Auth is required. Request is not authorized.
-                throw new UnauthorizedAccessException();
+                // Check if the activity is for a skill call and is coming from the Emulator.
+                if (activity.ChannelId == Channels.Emulator && activity is Activity act && act.RelatesTo != null)
```

Comment:
This seems like an obscure comparison... assuming if RelatesTo is not null means the the activity is a skill.  Is this defined by some underlying protocol behavior?

Reply:
I agree that this is obscure and that's why we are keeping internal. In the absence of a token, the only way we currently have of knowing if an activity is coming from a channel or another bot is the RelatesTo property. 
We discussed with @Stevenic and skills are the only ones using RelatestTo at the moment. 
We should consider updating the Activity Schema to add a new property to better inform the receiver if this is a skill call or a channel call, once we do that we can update the internal implementation to use the new property (but AuthenticateRequest will not change since the change will be internal)

Reply:
@EricDahlvang We have a meeting next week with Dan and others to discuss the final way of doing this. We hope we can do all in R11 but we should evaluate the trade-offs in the case we dn't get those changes in. The challenge is, given an incoming activity, determine if I am a skill at the time or not. RelatesTo is the only thing that we reliably set that seems to determine that. Now, with that and assuming the final protocol solution is not ready for R1, we have 2 options: 1) do this, and allow bot developers and composer users to use skills unauthenticated, knowing that it will work on the vast majority of scenarios but there is a bit of risk  or 2) not do this since it is a bit obscure.

Reply:
I agree with @EricDahlvang after checking the spec. Null-checking a property that has a complex object doesn't seem safe.

I unfortunately don't remember why the SDK populates the RelatesTo field and there weren't any comments in BFHttpClient explaining why it behaves like this. If someone wanted to obfuscate the user data before it was sent to the skill, they either have to mangle a cloned activity before calling the BFHttpClient or subclass the client and remove the logic. This feels like it should have been an opt-in feature.

Reply:
FYI, after reviewing with Dan and others we decided to use Role instead of ReleatesTo to check if an activity was sent by a skill or not. The changes are in my latest commit to this PR and I also updated the description the PR to summarize the changes I made.

Are we good here now?